packages/arb-token-bridge-ui/src/hooks/useCompleteMultiChainTransactions.ts

```diff
@@ -0,0 +1,338 @@
+import { useLatest } from 'react-use'
+import { useAccount } from 'wagmi'
+import useSWRImmutable from 'swr/immutable'
+
+import { ChainId, rpcURLs } from '../util/networks'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { getWagmiChain } from '../util/wagmi/getWagmiChain'
+import { useCallback, useEffect, useState } from 'react'
+import { fetchWithdrawalList } from '../util/withdrawals/fetchWithdrawals'
+import { fetchDepositList } from '../util/deposits/fetchDeposits'
+import {
+  L2ToL1EventResultPlus,
+  WithdrawalInitiated
+} from './arbTokenBridge.types'
+import { Transaction } from './useTransactions'
+import { MergedTransaction } from '../state/app/state'
+import {
+  getStandardizedTimestamp,
+  transformDeposit,
+  transformWithdrawal
+} from '../state/app/utils'
+import {
+  EthWithdrawal,
+  isTokenWithdrawal,
+  mapETHWithdrawalToL2ToL1EventResult,
+  mapTokenWithdrawalFromEventLogsToL2ToL1EventResult,
+  mapWithdrawalToL2ToL1EventResult
+} from '../util/withdrawals/helpers'
+import { FetchWithdrawalsFromSubgraphResult } from '../util/withdrawals/fetchWithdrawalsFromSubgraph'
+import { updateAdditionalDepositData } from '../util/deposits/helpers'
+
+export const PAGE_SIZE = 100
+
+export type AdditionalProperties = {
+  direction: 'deposit' | 'withdrawal'
+  source: 'subgraph' | 'event_logs'
+  parentChainId: ChainId
+  chainId: ChainId
+}
+
+export type Deposit = Transaction
```

Comment:
Maybe we can change `Transaction` to `Deposit`, I will see later when I remove more code